balancer/rls/control_channel.go

```diff
@@ -131,6 +133,17 @@ func (cc *controlChannel) dialOpts(bOpts balancer.BuildOptions, serviceConfig st
 	return dopts, nil
 }
 
+type ccStateSubscriber struct {
+	state *buffer.Unbounded
+}
+
+func (c *ccStateSubscriber) OnMessage(msg any) {
+	st, ok := msg.(connectivity.State)
+	if !ok {
+		return // Ignore invalid messages
```

Comment:
This is an error we don't expect to happen in practice and if it does, it indicates a severe programming error. I would be OK to add a `panic` here that includes the type being received.